types/underscore/underscore-tests.ts

```diff
@@ -1,3 +1,324 @@
+        // $ExpectType _Chain<SimpleStringObject, Dictionary<SimpleStringObject>>
+        result = _.chain(simpleStringObjectDictionary);
+
+        // $ExpectType _Chain<SimpleStringObject, Dictionary<SimpleStringObject>>
+        result = _<SimpleStringObject, _.Dictionary<SimpleStringObject>>(simpleStringObjectDictionary).chain();
+        // $ExpectType _Chain<SimpleStringObject, Dictionary<SimpleStringObject>>
+        result = _(simpleStringObjectDictionary).chain();
+
+    }
+
+    {
+        let result: _._Chain<string, string>;
+
+        // $ExpectType _Chain<string, string>
+        result = _.chain<string, string>(simpleString);
+        // $ExpectType _Chain<string, string>
+        result = _.chain(simpleString);
+
+        // $ExpectType _Chain<string, string>
+        result = _<string, string>(simpleString).chain();
+        // $ExpectType _Chain<string, string>
+        result = _(simpleString).chain();
+    }
+
+    {
+        let result: _._Chain<number, number>;
+
+        // $ExpectType _Chain<number, number>
+        result = _.chain<number>(simpleNumber);
+        // $ExpectType _Chain<number, number>
+        result = _.chain(simpleNumber);
+
+        // $ExpectType _Chain<number, number>
+        result = _<number>(simpleNumber).chain();
+        // $ExpectType _Chain<number, number>
+        result = _(simpleNumber).chain();
+    }
+}
+
+// value
+// verify that the object given to chain is returned by value
+{
+    {
+        let result: SimpleStringObject[];
+
+        // $ExpectType SimpleStringObject[]
+        result = _.chain<SimpleStringObject, SimpleStringObject[]>(simpleStringObjectArray).value();
+        // $ExpectType SimpleStringObject[]
+        result = _.chain(simpleStringObjectArray).value();
+    }
+
+    {
+        let result: _.List<SimpleStringObject>;
+
+        // $ExpectType List<SimpleStringObject>
+        result = _.chain<SimpleStringObject, _.List<SimpleStringObject>>(simpleStringObjectList).value();
+        // $ExpectType List<SimpleStringObject>
+        result = _.chain(simpleStringObjectList).value();
+    }
+
+    {
+        let result: _.Dictionary<SimpleStringObject>;
+
+        // $ExpectType Dictionary<SimpleStringObject>
+        result = _.chain<SimpleStringObject, _.Dictionary<SimpleStringObject>>(simpleStringObjectDictionary).value();
+        // $ExpectType Dictionary<SimpleStringObject>
+        result = _.chain(simpleStringObjectDictionary).value();
+
+    }
+
+    {
+        let result: string;
+
+        // $ExpectType string
+        result = _.chain<string, string>(simpleString).value();
+        // $ExpectType string
+        result = _.chain(simpleString).value();
+    }
+
+    {
+        let result: number;
+
+        // $ExpectType number
+        result = _.chain<number>(simpleNumber).value();
+        // $ExpectType number
+        result = _.chain(simpleNumber).value();
+    }
+}
+
+// Additional Tests
```

Comment:
Are you planning a wholesale cleanup of the test file? Otherwise, please append rather than prepend the new tests and add a commit to equalize the identation.

Reply:
I will be adding a large set of tests that may make some of these tests obsolete, but I've so far left them alone to show that I'm not breaking any of them. I'd be happy to move my additions to the end while this PR chain is in progress, but it might be worth revisiting that at the end of this set of PRs to do some cleanup of redundant existing tests and possibly also reorder the tests since I feel like the ones I'm adding may be better examples of comprehensive tests that it might be preferable for future editors to encounter first when looking at this file.

Reply:
I'm not sure what you mean by "equalize the indentation;" would you mind clarifying?

Reply:
You are using four-space indents. The rest of the file uses 2-space indents.

Reply:
Scratch that, I don't know why I thought that.

Reply:
Sounds good; I've moved my changes to the bottom of the file and will continue to add them there in future PRs.